src/profile-logic/marker-schema.js

```diff
@@ -671,7 +671,7 @@ export function markerPayloadMatchesSearch(
   for (const payloadField of markerSchema.data) {
     if (payloadField.searchable) {
       let value = data[payloadField.key];
-      if (payloadField.format === 'unique-string') {
+      if (value && payloadField.format === 'unique-string') {
```

Comment:
We really need that you check explicitely for `undefined`.
Indeed we want to handle optional values.
Note that for example `0` is a valid string id, and checking just for `if (value)` will skip it in that case.

If we want to be resilient for invalid values, we can do something like this:

```js
if (value === undefined || value === null) {
  continue;
}

if (payloadField.format === 'unique-string') {
  if (typeof value !== "number") {
    console.error(`In marker ${marker.name}, the key ${payloadField} has an invalid value ${value} as a unique-string.`);
    continue;
  }
  if (!stringTable.hasIndex(value)) {
    console.error(`In marker ${marker.name}, the key ${payloadField} has an invalid index ${value} as a unique-string.`);
    continue;
  }
  value = stringTable.getString(value);
}

if (value === '') {
  continue;
}
```

Reply:
Got it. Making the change.

Reply:
@julienw 
```
We really need that you check explicitely for undefined.
Indeed we want to handle optional values.
Note that for example 0 is a valid string id, and checking just for if (value) will skip it in that case.
```

Can we use nullish coalescing operator like this:
```
let value = data[payloadField.key] ?? false;
if(!value) continue;
````

So we can replace this two blocks by only one check.
```
if (value === undefined || value === null) {
  continue;
}
// other codes
if (value === '') {
  continue;
}
```

Reply:
The nullish coalescing operator can be very useful, but this doesn't work here, because the code between the 2 checks can also assign to "value". So the second if checks also for the value returned from the string table.